src/main/java/world/bentobox/challenges/database/object/requirements/IslandRequirements.java

```diff
@@ -23,202 +29,235 @@
-// ---------------------------------------------------------------------
-
-
-	/**
-	 * Method isValid returns if given requirement data is valid or not.
-	 *
-	 * @return {@code true} if data is valid, {@code false} otherwise.
-	 */
-	@Override
-	public boolean isValid()
-	{
-		return super.isValid() &&
-			this.requiredBlocks != null && this.requiredBlocks.keySet().stream().noneMatch(Objects::isNull) &&
-			this.requiredEntities != null && this.requiredEntities.keySet().stream().noneMatch(Objects::isNull);
-	}
-
-
-	/**
-	 * Method Requirements#copy allows copies Requirements object, to avoid changing content when it is necessary
-	 * to use it.
-	 * @return IslandRequirements copy
-	 */
-	@Override
-	public Requirements copy()
-	{
-		IslandRequirements clone = new IslandRequirements();
-		clone.setRequiredPermissions(new HashSet<>(this.getRequiredPermissions()));
-
-		clone.setRequiredBlocks(new HashMap<>(this.requiredBlocks));
-		clone.setRemoveBlocks(this.removeBlocks);
-		clone.setRequiredEntities(new HashMap<>(this.requiredEntities));
-		clone.setRemoveEntities(this.removeEntities);
-
-		clone.setSearchRadius(this.searchRadius);
-
-		return clone;
-	}
-
-
-// ---------------------------------------------------------------------
-// Section: Variables
-// ---------------------------------------------------------------------
-
-
-	/**
-	 * Map that contains which materials and how many is necessary around player to complete challenge.
-	 */
-	@Expose
-	private Map<Material, Integer> requiredBlocks = new EnumMap<>(Material.class);
-
-	/**
-	 * Boolean that indicate if blocks should be removed from world after completion.
-	 */
-	@Expose
-	private boolean removeBlocks;
-
-	/**
-	 * Map that contains which entities and how many is necessary around player to complete challenge.
-	 */
-	@Expose
-	@JsonAdapter(EntityCompatibilityAdapter.class)
-	private Map<EntityType, Integer> requiredEntities = new EnumMap<>(EntityType.class);
-
-	/**
-	 * Boolean that indicate if entities should be removed from world after completion.
-	 */
-	@Expose
-	private boolean removeEntities;
-
-	/**
-	 * Radius for searching distance for blocks and entities.
-	 */
-	@Expose
-	private int searchRadius = 10;
+    /**
+     * Constructor Requirements creates a new Requirements instance.
+     */
+    public IslandRequirements() {
+        // Empty constructor for data loader
+    }
+
+    // ---------------------------------------------------------------------
+    // Section: Variables
+    // ---------------------------------------------------------------------
+
+
+    /**
+     * Map that contains which materials and how many is necessary around player to complete challenge.
+     */
+    @Expose
+    private Map<Material, Integer> requiredBlocks = new EnumMap<>(Material.class);
+
+    @Expose
+    private Map<Tag<Material>, Integer> requiredMaterialTags = new HashMap<>();
+    @Expose
+    private Map<Tag<Fluid>, Integer> requiredFluidTags = new HashMap<>();
```

Comment:
New fluid tag map added, but this is not used right now. Could be used in the future maybe.